airflow/models/dagbag.py

```diff
@@ -151,10 +151,10 @@ def get_dag(self, dag_id):
                 # Load from DB if not (yet) in the bag
                 self._add_dag_from_db(dag_id=dag_id)
```

Comment:
We can return here -- if we've just loaded it we don't need to do any of these checks, do we?

Reply:
Updated in https://github.com/apache/airflow/pull/9851/commits/02856d63a5d500788c8aab7c49a235b036800364